src/main/scala/mesosphere/marathon/core/task/termination/impl/KillServiceActor.scala

```diff
@@ -66,8 +66,11 @@ private[impl] class KillServiceActor(
   }
 
   override def preStart(): Unit = {
-    context.system.eventStream.subscribe(self, classOf[InstanceChanged])
+    // TODO: this actor needs to initialized once with all instances with goals Stopped/Decommissioned.
+
     context.system.eventStream.subscribe(self, classOf[UnknownInstanceTerminated])
+    context.system.eventStream.subscribe(self, classOf[InstanceChanged])
```

Comment:
What do you think about subscribing to `InstanceTracker.instanceUpdates` instead?

Reply:
what would be the advantage?

Reply:
@timcharper mentioned this somewhere. If I remember correctly the stream is more performant.